### PR TITLE
Adds --minimal option for running MooseApp w/o an input file.

### LINKED ISSUE #6126: Support creation of minimal, no-solve application with Multiapps
Per our conversation with @rppawlo:

We need the ability to create a minimal no-solve top level application with multiapps that can be appropriately controlled from an external application such as Vera.

### PR DESCRIPTION
**Pull request by aeslaughter**
(closes #6126)

### CODE REVIEW

File: unit/src/MinimalApp.C [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "MinimalApp.h"
+#include "AppFactory.h"
+#include "Executioner.h"
+#include "MooseMesh.h"
+
+CPPUNIT_TEST_SUITE_REGISTRATION( MinimalApp );
+
+void
+MinimalApp::createMinimalAppTest()
```
Comment (permcody) [inline unit/src/MinimalApp.C]: This is great!

**Comment by permcody** [comment]
@rppawlo - Please take a look at this PR. We are demonstrating how to build a complete minimal application from an external application (we have a unit test to show you how). Hopefully this will work for you, if not let us know how to improve on this design.

Thanks!

**Comment by rppawlo** [comment]
@permcody Wow!  Looks great!  I will work on this tomorrow and get back to you. Thanks for the fast turn around! Really appreciate it!

When do you think this will be merged into the main repo?

**Comment by permcody** [comment]
You can thank @aeslaughter! It looks good from my end so if you don't need
any changes we can get it merged immediately. I just wanted to make sure
you had a chance to review before we merge it.

**Comment by rppawlo** [comment]
Thanks @aeslaughter ! I will merge these changes into my local bison repo and try to do an update in casl to see how far we can get. I don't see anything obvious that needs to change at this point.

**Pull request merged**